providers/implementations/macs/kmac_prov.c

```diff
@@ -340,6 +360,14 @@ static int kmac_final(void *vmacctx, unsigned char *out, size_t *outl,
     if (!ossl_prov_is_running())
         return 0;
 
+#ifdef FIPS_PROVIDER
+    if (FIPS_kmac_length_checks_enabled(PROV_LIBCTX_OF(kctx->provctx))
```

Comment:
Ideally we would be adding tests for these paths..

Reply:
I've added failure tests.  There isn't a need to add success ones IMO since that's already done.